blogs/2025/07/16/copilot-coding-agent.md

```diff
@@ -0,0 +1,104 @@
+![PR from Coding Agent shown in VS Code with a screenshot](draft-with-screenshot.png)
+
+You can review the agent’s PR, leave comments, and ask for changes - all from within VS Code. The agent will pick up your feedback, update the PR, and let you know when it's finished.
+
+This workflow integrates beautifully with services that offer previews for pull requests.
+For example, if you use Azure Static Web Apps, Vercel, Netlify, and others, you can review the agent's work without even having to check out and run a branch locally.
+
+![Live preview of the site on Netlify but we have an error](live-preview-netlify.png)
+
+Uh oh - now that we look closely, it looks like we didn't quite get the "Trending" section working. In this case, that's ok. In fact it's good. Our agent couldn't succeed in one shot because it doesn't have access to create the required database changes. Instead, it created a migration script as part of the PR. That *will* require us to manually check out the PR, but you might feel more comfortable being in control over that sort of change.
+
+But what if we *did* want to give access to make the database changes to a dev environment? Coding Agent has support for MCP servers, and provided you have one for your database of choice (in this case Supabase), you can [configure MCP servers for the agent on the repo](https://docs.github.com/en/copilot/how-tos/agents/copilot-coding-agent/extending-copilot-coding-agent-with-mcp). It's up to you - how autonomous do you want the agent to be?
+
+![MCP configuration shown for Copilot Coding Agent on github.com](coding-agent-mcp-config.png)
+
+But it gets even better - because you can do all of this right from VS Code.
+
+### Start sessions from Copilot Chat
+
+We find ourselves wanting to do more and more right from the [chat panel](https://code.visualstudio.com/docs/copilot/chat/copilot-chat) in VS Code. This is where most work and thinking happens and you're not locked into the issue workflow. You can - at any point in a chat, no matter how long - delegate the task directly to the Coding Agent. All of the context that is currently a part of that chat will get sent to the Coding Agent.
+
+The Coding Agent then opens a PR and goes directly to work - skipping the issue step entirely. It distills down the context into a detailed PR description and you'll see that Copilot Coding Agent makes use of "to-do lists" so that you can see what it's doing and where it is in the process at a glance.
+
+<video src="delegate-to-coding-agent.mp4" title="Delegate to Coding Agent video" controls poster="delegate-to-coding-agent.png"></video>
+
+## What’s next?
+
+The Copilot Coding Agent integration in Visual Studio Code is currently in preview, but you can use it all today. If you want to light all of this up in your editor, all you need to do is make sure you have the GitHub Pull Requests extension installed and add the following setting...
+
+```json
+"githubPullRequests.codingAgent.uiIntegration": true,
+```
```

Comment:
I just realized this is at the very end - but a lot of people won't make it this far. We gotta make this the "call to action" and bring it to the very top. The flow is

1. We've got a sneak peek of something big - but you can use it today.
2. Install this extension and add this one setting, and check out what you get
3. \[Rest of blog post]

Reply:
We can repeat this at the end in case people missed it - "as a reminder, this is in preview, and you'll need to enable `"githubPullRequests.codingAgent.uiIntegration": true,`

Reply:
added a mention at the top, lmk what you think